### ISSUE TITLE
Make use of translatable config string

### ISSUE THREAD

**Issue by robertgarrigos** [backdrop/backdrop-issues#6210]
## Description of the bug

Core use module has some config strings defined, in user.mail.json, as translatable. But they are never used as such. Thus, you can find those strings through the translation UI, but their translations are never used.

## Steps To Reproduce

* Add a second language to a backdrop installation
* Set second language as default
* Translate any of the strings in core/modules/user/config/user.mail.json
* Look for the translations on /admin/config/people/emails: the strings are not being translated

## Actual behavior

The config fields in /admin/config/people/emails are shown only in English

## Expected behavior

Once translated, they should be shown ther

## Additional information

user_settings_email() function in user.admin.inc sets the default values for the form shown on /admin/config/people/emails with the config values without any translation. Doing it by using config_get_translated() function would resolve this problem.

**Comment by robertgarrigos** [comment]
This [PR](https://github.com/backdrop/backdrop/pull/4506) is ready to be tested/reviewed.

**Comment by argiepiano** [comment]
This PR will indeed show the translated strings in the UI form when you switch your language. But the issue with this approach is that, when you submit the form, the **translated version** will be saved to the config file. The config file should only include the **original, untranslated** string to allow for the `locale()` function to do its job translating it to the current language.

This seems to be a shortcoming of the way config translations are handled in Backdrop. In this case, the translations do work when retrieving the value for sending the email, but the UI form to set up the user mail texts is not prepared to handle the translated strings when they are shown in the form elements.

In Drupal 7, these user mail variables are translatable ONLY if you install i18n_variable. Once you do that and enable the translations of those specific variables, you get this language switcher for the UI form that allows you to handle translations:

<img width="626" alt="Screen Shot 2023-08-30 at 4 56 09 PM" src="https://github.com/backdrop/backdrop-issues/assets/9938978/414fc3c2-5a0d-4e95-971b-64f7bb848a97">

So, we need something like this in core to handle these translation. Adding a feature like this is complicated, given that these are form-specific. So, one could add this to the user mail form, but what about all the other UI forms that are used to set config values? 

In short, this PR doesn't do that, unfortunately, and actually it introduces a regression by saving the translated string to the config file.

**Comment by robertgarrigos** [comment]
True. :-(
